packages/beacon-node/src/network/peers/score/score.ts

```diff
@@ -46,6 +46,16 @@ export class RealScore implements IPeerScore {
     this.setLodestarScore(newScore);
   }
 
+  applyReconnectionCoolDown(timeInMin: number): void {
```

Comment:
prefer no param here and define `RECONNECTION_COOL_DOWN_BEFORE_DECAY_MS` instead, similar to `BANNED_BEFORE_DECAY_MS`. The method name `applyReconnectionCoolDown` already gives us the context

the scoring system should decides the score on its own instead of let consumers decide it

Reply:
as noted in the other comment we also need something like `applyTooManyPeersCoolDown` `applyIrreleventNetwork ` etc

we may just need `applyCoolDown` with a reason enum, based on each each enum we'll have a `*BEFORE_DECLAY_MS` for it